### PR TITLE
Add "insecure by whitelisting" state

### LINKED ISSUE #280: Different tooltip for manual whitelist opt-in
### Idea

When the server's detected status is unintrusive, but the player manually opts into whitelisting it (in the server, by clicking the icon), there should be a special message describing the state thoughout the session. 

Could be something like `gui.nochatreports.status_insecure_whitelist` that says that the user itself opted into the insecure state without the server requiring it.

### Reasoning

I am aware that once the player rejoins the server, the message will revert to `gui.nochatreports.status_insecure` again due to technical reasons, but at least the player would see the authentic state at the time of toggling the whitelist, which in my opinion is better than nothing.

### Other Information

1.19.3+ only

### PR DESCRIPTION
**Pull request by Madis0**
Fixes #280

### CODE REVIEW

File: src/main/java/com/aizistral/nochatreports/mixins/client/MixinChatScreen.java [hunk 1]
```diff
@@ -86,7 +86,12 @@ private void onInit(CallbackInfo info) {
 
 								if (ServerSafetyState.getCurrent() != ServerSafetyLevel.SECURE) {
 									ServerSafetyState.setAllowChatSigning(true);
-									ServerSafetyState.updateCurrent(ServerSafetyLevel.INSECURE);
+									if(ServerSafetyState.getCurrent() == ServerSafetyLevel.INSECURE){
+										ServerSafetyState.updateCurrent(ServerSafetyLevel.INSECURE);
+									}
```
Comment (Aizistral) [inline src/main/java/com/aizistral/nochatreports/mixins/client/MixinChatScreen.java]: I don't quite follow the logic here, why call `updateCurrent` if state is already set to `INSECURE`?

**Review by Aizistral — COMMENTED** [review]
Overall I don't think it would hurt to merge this for now. But keep in mind that I plan to redesign user's options around signing in near future, and I'm not sure how this logic would fit into the new paradigm, so we're probably gonna have to dump this particular implementation in the end.

File: src/main/java/com/aizistral/nochatreports/mixins/client/MixinChatScreen.java (hunk 1 above)
Comment (Madis0) [reply src/main/java/com/aizistral/nochatreports/mixins/client/MixinChatScreen.java]: Honestly, I don't know. 
I just looked at how you did it at the moment and thought that maybe I would need to keep the line to make it "re-updated to reflect the state" or something. Can be removed if that's not the case.

Comment (Aizistral) [reply src/main/java/com/aizistral/nochatreports/mixins/client/MixinChatScreen.java]: Yeah that's not needed, just remove it.

**Review by Aizistral — APPROVED** [review]

**Pull request merged**